tools/SourceKit/tools/sourcekitd/include/sourcekitd/Internal.h

```diff
@@ -128,11 +128,17 @@ class RequestDict {
   sourcekitd_object_t Dict;
 
 public:
-  explicit RequestDict(sourcekitd_object_t Dict) : Dict(Dict) {}
-
-  sourcekitd_uid_t getUID(SourceKit::UIdent Key);
-  Optional<llvm::StringRef> getString(SourceKit::UIdent Key);
-  Optional<RequestDict> getDictionary(SourceKit::UIdent Key);
+  explicit RequestDict(sourcekitd_object_t Dict) : Dict(Dict) {
+    sourcekitd_request_retain(Dict);
+  }
+  RequestDict(const RequestDict &other) : RequestDict(other.Dict) {}
+  ~RequestDict() {
+    sourcekitd_request_release(Dict);
+  }
```

Comment:
This may trigger unnecessary retain/release traffic, but this should make the request dictionary handling safer.